course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+                       ['bob', {user_id: scope.selectedResult.id,
+                                role_id: scope.selectedRole.roleId}]);
+           }
+        );
+    });
+    describe('addUserToCourse', function() {
+        var user = {user_id: 'bobdobbs', role_id: 0};
+        var searchTerm = 'bob_dobbs@harvard.edu';
+        var enrollmentDetailsURL = coursePeopleURL + '&user_id=' + user.user_id;
+        var enrollmentDetails = {
+            results: [{
+                profile: {
+                    name_last: 'Dobbs',
+                    name_first: 'Bob',
+                    role_type_cd: 'STUDENT',
+                    univ_id: 'bobdobbs',
+                },
+                role: {
+                    role_name: 'Student',
+                }
+            }],
+        };
+
+        beforeEach(function() {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'addUserToCourse test',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+            scope.searchInProgress = true;
+            scope.searchResults = [{}];
+        });
+
+        afterEach(function() {
+            // no matter what, we should end the search and clear the results.
+            expect(scope.searchInProgress).toBe(false);
+            expect(scope.searchResults).toEqual([]);
+        });
+
+        it('should alert and reload the datatable on success', function() {
+            var expectedSuccess = 
+                JSON.parse(JSON.stringify(enrollmentDetails.results[0]));
+            expectedSuccess.searchTerm = searchTerm;
+
+            // mock out the datatable so we can verify that it gets reloaded
+            scope.dtInstance = {reloadData: function(){}};
+            spyOn(scope.dtInstance, 'reloadData');
+
+            // call it
+            scope.addUserToCourse(searchTerm, user);
+
+            // trigger the ajax calls
+            $httpBackend.expectPOST(coursePeopleURL, user)
+                        .respond(201, JSON.stringify(user)); 
+            $httpBackend.expectGET(enrollmentDetailsURL)
+                        .respond(200, JSON.stringify(enrollmentDetails));
+            $httpBackend.flush(2);
+
+            // check to see if it's reacting correctly
+            expect(scope.successes).toEqual([expectedSuccess]);
+            expect(scope.dtInstance.reloadData.calls.count()).toEqual(1);
+        });
+
+        it('should alert on a partial failure', function() {
+            var partialFailureResponse = 
+                JSON.parse(JSON.stringify(enrollmentDetails.results[0]));
+            partialFailureResponse.detail = 'Some kind of partial failure';
+            var expectedSuccess = 
+                JSON.parse(JSON.stringify(enrollmentDetails.results[0]));
+            expectedSuccess.searchTerm = searchTerm;
```

Comment:
Hrm...when I went to do this, I noticed a bunch of other `expectedX`-type definitions that I could move up as well.  Moving them all away means the test data is further away from the test code than I'd like, and it feels like it hurts readability.  Think I'll pass on changing this, let me know if you feel strongly about it.